### PR TITLE
feat(replica_vol): adding owner id attribute to replica

### PR DESCRIPTION
**Pull request by abhilashshetty04**

### CODE REVIEW

File: apis/io-engine/protobuf/v1/replica.proto [hunk 1]
```diff
@@ -16,6 +16,13 @@ service ReplicaRpc {
   rpc ShareReplica (ShareReplicaRequest) returns (Replica) {}
   rpc UnshareReplica (UnshareReplicaRequest) returns (Replica) {}
   rpc ResizeReplica (ResizeReplicaRequest) returns (Replica) {}
+  rpc SetReplicaOwner (SetReplicaOwnerRequest) returns (Replica) {}
+}
+
+// Request struct to set the owner of the replica, If not already set.
+message SetReplicaOwnerRequest {
+  string uuid = 1;  // Uuid of the replica.
+  string owner_id = 2; // Owner of the replica.
```
Comment (tiagolobocastro) [inline apis/io-engine/protobuf/v1/replica.proto]: Might be worth using entity nomenclature which is used by the snapshot already?

```suggestion
message SetReplicaEntityRequest {
  string uuid = 1;  // Uuid of the replica.
  string entity_id = 2; // Id of the entity which owns the replica.
```

**Review by tiagolobocastro — APPROVED** [review]
Please squash

File: apis/io-engine/protobuf/v1/replica.proto [hunk 2]
```diff
@@ -34,6 +41,7 @@ message ReplicaSpaceUsage {
 message Replica {
   string name = 1;  // name of the replica
   string uuid = 2;  // uuid of the replica
+  optional string entity_id = 14; // entity which owns the replica, if any
```
Comment (hrudaya21) [inline apis/io-engine/protobuf/v1/replica.proto]: Change the position of the new field. Should be as per its index.

File: apis/io-engine/protobuf/v1/replica.proto [hunk 3]
```diff
@@ -50,6 +58,7 @@ message Replica {
 message CreateReplicaRequest {
   string name = 1;  // name of the replica
   string uuid = 2;  // uuid of the replica
+  optional string entity_id = 14; // entity which owns the replica, if any
```
Comment (hrudaya21) [inline apis/io-engine/protobuf/v1/replica.proto]: Here also position can be changed to last.

**Review by datacore-vvarakantham — APPROVED** [review]

**Review by hrudaya21 — APPROVED** [review]

File: apis/io-engine/protobuf/v1/replica.proto [hunk 4]
```diff
@@ -55,6 +63,7 @@ message CreateReplicaRequest {
   bool thin = 5;    // thin provisioning
   ShareProtocol share = 6;  // protocol to expose the replica over
   repeated string allowed_hosts = 7; // host (nqn's) which are allowed to connect to the target.
+  optional string entity_id = 14; // entity which owns the replica, if any
```
Comment (tiagolobocastro) [inline apis/io-engine/protobuf/v1/replica.proto]: SHould be 8

**Review by tiagolobocastro — CHANGES_REQUESTED** [review]

**Review by tiagolobocastro — APPROVED** [review]

**Pull request merged**